ingestion-beam/bin/download-document-sample

```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env python3
+
+"""Download samples of json documents from the decoded and error stream.
+
+This is meant for integration testing, and is easily inspected through the command-line.
+For example, to count the total number documents per group:
+
+  cat document_sample.ndjson | \
+  jq -rc '.attributeMap | [.document_namespace, .document_type, .document_version]' | \
+  uniq -c
+"""
+
+import argparse
+import base64
+import gzip
+import json
+import logging
+import os
+import time
+
+from google.cloud import bigquery
+
+INGESTION_BEAM_ROOT = os.path.realpath(
+    os.path.join(os.path.dirname(os.path.realpath(__file__)), "..")
+)
+
+# formatted using the BigQuery console formatter
+QUERY = """
+-- Create a PubSub compatible row with the most recent document samples that
+-- have been decoded.
+SELECT
+  STRUCT( document_namespace,
+    document_type,
+    document_version ) AS attributeMap,
+  payload
+FROM
+  `moz-fx-data-shared-prod`.monitoring.document_sample_nonprod_v1
+WHERE
+  document_decoded
+  AND submission_timestamp = (
+  SELECT
+    MAX(submission_timestamp)
+  FROM
+    `moz-fx-data-shared-prod`.monitoring.document_sample_nonprod_v1 )
+ORDER BY
+  document_namespace,
+  document_type,
+  document_version
+"""
+
+
+def extract_samples(query):
+    """A generator for reading documents in the sampled landfill dataset that exist in the schemas.
```

Comment:
We should probably avoid the "landfill" jargon now. We've been tending to call this data "payload_bytes" since that's how the tables are named.

Reply:
Sorry, this was missed during the rewrite of `download-sampled-landfill`, I've tried to be consistent with `payload_bytes_(decoded|error)` and "document sample" wording.